downloader.go

```diff
@@ -174,42 +176,87 @@ func (dl *Downloader) listAvailableFiles(audioLibrary *AudioFileLibrary, referen
 	return availableFiles
 }
 
-// Check that the sound file is valid.  This may mean checking it's size on disk compared
-// to the info the API server sends us, or even it's file hash.
-func (dl *Downloader) validateSoundFile(audioLibrary *AudioFileLibrary, fileID int) bool {
-	// Just return true for now.
+// Check that the sound file is valid.
+func (dl *Downloader) validateSoundFile(fileNameOnDisk string, fileSize int) bool {
+
+	// Check size on disk is the same as the size the api-server tells us this file should be.
+	file, err := os.Open(fileNameOnDisk)
+	if err != nil {
+		return false
+	}
+
+	fileInfo, err := file.Stat()
```

Comment:
I think you want `os.Stat` instead of calling `Stat` on a file instance. Then there's no need to open the file.